Library/ViewModels/LiveStreamEventDetailsViewModel.swift

```diff
@@ -82,7 +82,20 @@ public final class LiveStreamEventDetailsViewModel: LiveStreamEventDetailsViewMo
       )
       .map(first)
 
-    let event = configData.map(second)
+    let initialEvent = configData.map(second)
+
+    let updatedEventFetch = initialEvent
+      .switchMap { event -> SignalProducer<Event<LiveStreamEvent, LiveApiError>, NoError> in
+        AppEnvironment.current.liveStreamService
+          .fetchEvent(eventId: event.id, uid: AppEnvironment.current.currentUser?.id)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+    }.take(first: 1)
```

Comment:
you can also `prefix(value: event)` in the `switchMap`, and then the signal below just becomes `let event = updatedEventFetch.values()`